### ISSUE TITLE
False positive NP_NONNULL_RETURN_VIOLATION when using JDT @NonNullByDefault and @Nullable annotations

### ISSUE THREAD

**Issue by asselmeyer** [spotbugs/spotbugs#805]
This issue seems to be related to #562 and #471. When using Eclipses JDT `@NonNullByDefault` annotation on a package-info.class and specifying a null-returning method as `@Nullable` in that package, SpotBugs (version 3.1.8) reports a NP_NONNULL_RETURN_VIOLATION on that method.

It seems that SpotBugs recognizes the `@NonNullByDefault` annotation but not the `@Nullable` annotation. As a consequence SpotBugs reports the error for methods that return `null` as these are falsely assumed to be annotated as `@Nonnull`.

See https://github.com/asselmeyer/spotbugs-default-annotation-bug for a minimal example illustrating this issue. In a nutshell in looks like this:
`package-info.java`
```java
@org.eclipse.jdt.annotation.NonNullByDefault
package sample.withdefault;
```
`NullableSample.java`:
```java
package sample.withdefault;

public class NullableSample {

	@Nullable
	public Object maybeNull() {
		return null;
	}
}
``` 

SpotBugs then reports an error on the `maybeNull` method:
`spotbugs.xm`
```xml
<?xml version='1.0' encoding='UTF-8'?>
<BugCollection version='3.1.8' threshold='low' effort='default'>
    <file classname='sample.withdefault.NullableSample'>
        <BugInstance type='NP_NONNULL_RETURN_VIOLATION' priority='High' category='CORRECTNESS' message='sample.withdefault.NullableSample.maybeNull() may return null, but is declared @Nonnull' lineNumber='9' />
    </file>
    ...
</BugCollection>
```

**Comment by vrotaru** [comment]
One more problem is that `@Nullable` in this setup is also ignored for parameters. If I change the `NullableSample` to

```Java
public class NullableSample {

    public void maybeNull(String foo, @Nullable String bar) {
		System.out.println(foo + bar);
    }

    public void useOfNull(String foo) {
        maybeNull(foo, null);
    }

}
```

and then run 

```Sh
mvn clean compile spotbugs:check
```

I get the following error
```Console
[INFO] --- spotbugs-maven-plugin:3.1.8:check (default-cli) @ spotbugs-default-annotation-bug ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[ERROR] Null passed for non-null parameter of maybeNull(String, String) in sample.withdefault.NullableSample.useOfNull(String) [sample.withdefault.NullableSample] At NullableSample.java:[line 12] NP_NONNULL_PARAM_VIOLATION
[INFO] 
```

**Comment by KengoTODA** [comment]
@vrotaru could you check #633, maybe you need to use `@CheckForNull` instead of `@Nullable`.

**Comment by vrotaru** [comment]
@KengoTODA Nope, that's a different issue. 

Besides I cannot use `@CheckForNull` since it is provided by `jsr305` jar (artifact) not the Eclipse `jdt-annotations`

**Comment by nngo** [comment]
The Eclipse JDT `@NonNullByDefault` annotation allows locations to be specified, so I get a false positive that a nullable field should be non-null..

```java
@NonNullByDefault(DefaultLocation.RETURN_TYPE)
public class Foo {
  private String name;

  public Foo() { //<- got NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR 
  }
}
```

**Comment by enexusde** [comment]
Please notice that @NotNullByDefault may in `package-info.java` also.
